src/qs_tddfpt2_restart.F

```diff
@@ -287,5 +291,127 @@ FUNCTION tddfpt_read_restart(evects, evals, gs_mos, logger, tddfpt_section, tddf
+   SUBROUTINE tddfpt_write_newtonx_output(evects, evals, gs_mos, logger, tddfpt_print_section, qs_env)
+
+      TYPE(cp_fm_p_type), DIMENSION(:, :), INTENT(in)    :: evects
+      REAL(kind=dp), DIMENSION(:), INTENT(in)            :: evals
+      TYPE(tddfpt_ground_state_mos), DIMENSION(:), &
+         INTENT(in)                                      :: gs_mos
+      TYPE(cp_logger_type), POINTER                      :: logger
+      TYPE(section_vals_type), POINTER                   :: tddfpt_print_section
+      TYPE(qs_environment_type), INTENT(in), POINTER     :: qs_env
+
+      CHARACTER(LEN=*), PARAMETER :: routineN = 'tddfpt_write_newtonx_output'
+
+      INTEGER                                            :: handle, iocc, ispin, istate, ivirt, nao, &
+                                                            nspins, nstates, ounit
+      INTEGER, DIMENSION(maxspins)                       :: nmo_occ, nmo_virt
+      LOGICAL                                            :: print_virtuals
+      TYPE(cp_para_env_type), POINTER                    :: para_env
+
+      IF (BTEST(cp_print_key_should_output(logger%iter_info, tddfpt_print_section, "NAMD_PRINT"), cp_p_file)) THEN
+         CALL timeset(routineN, handle)
+         NULLIFY (para_env)
+         CALL get_qs_env(qs_env, para_env=para_env)
+
+         nspins = SIZE(evects, 1)
+         nstates = SIZE(evects, 2)
+
+         IF (debug_this_module) THEN
+            CPASSERT(SIZE(evals) == nstates)
+            CPASSERT(nspins > 0)
+            CPASSERT(nstates > 0)
+         END IF
+
+         CALL cp_fm_get_info(gs_mos(1)%mos_occ, nrow_global=nao)
+
+         ounit = cp_print_key_unit_nr(logger, tddfpt_print_section, "NAMD_PRINT", &
+                                      extension=".inp", file_form="FORMATTED", file_action="WRITE", file_status="REPLACE")
+
+         ! print eigenvectors
+         DO istate = 1, nstates
+            DO ispin = 1, nspins
+               ! TDDFPT wave function is actually stored as a linear combination of virtual MOs
+               ! that replaces the corresponding deoccupied MO. Unfortunately, the phase
+               ! of the occupied MOs varies depending on the eigensolver used as well as
+               ! how eigenvectors are distributed across computational cores. The phase is important
+               ! because TDDFPT wave functions are used to compute a response electron density
+               ! \rho^{-} = 1/2 * [C_{0} * evect^T + evect * C_{0}^{-}], where C_{0} is the expansion
+               ! coefficients of the reference ground-state wave function. To make the restart file
+               ! transferable, TDDFPT wave functions are stored in assumption that all ground state
+               ! MOs have a positive phase.
+               CALL cp_fm_column_scale(evects(ispin, istate)%matrix, gs_mos(ispin)%phases_occ)
+
+               IF (ounit > 0) THEN
+                  WRITE (ounit, "(A)") "                    "
```

Comment:
What is the point of these blanks?

Reply:
None, sorry, I will remove them.

Reply:
If you just want an empty line before a header use
```
WRITE (ounit, "(/,A)") "ES EIGENVECTORS SIZE"
```

Reply:
ok, thanks!

Reply:
I will recommit in a second

Reply:
I adjusted now, thanks :)